### ISSUE TITLE
Reconsider CI management

### ISSUE THREAD

**Issue by effigies** [nipy/nipype#3565]
> Looks like CI is broken; apologies. @ghisvail If you have time to look into this, it would be appreciated. Otherwise I'll get to it this week.

It looks like issues with CI started to surface on the last two merged PRs. CI complains that jobs no longer find `black` or `pytest` although they appear in the install step in the logs. The only reason I could think of is that the venv is no longer visible or activated between calls to the CI scripts (perhaps as a result of a change in GitHub Actions behaviour)?

Perhaps we could refactor CI to not use a dedicated venv? Maybe there are good reasons for it, I have just never seen it done in practice with GHA since each job is executed in a separate throwaway container anyway. It might also make things considerably simpler, as I find the current flow of CI scripts quite hard to follow. Let me know what you think.

_Originally posted by @ghisvail in https://github.com/nipy/nipype/issues/3529#issuecomment-1513199075_

**Comment by effigies** [comment]
Yeah, I now regret the `tools/ci` approach. The idea was to make it easier to switch between CIs by creating a minimum set of scripts that would work locally, so the learning curve for a new CI would just be "how do I run these scripts?". Now that I know GHA better, it's pretty confining... I think tox or hatch would be much better tools for that purpose.

I agree we can drop the venvs now. It used to be useful to avoid accidentally depending on your build tools in your test environment. `pyproject-build` has completely made that unnecessary.

**Comment by ghisvail** [comment]
Now that I am getting very familiar with Hatch, I could give it a go if I have got your green light for it?

**Comment by ghisvail** [comment]
> Have you found integrating Hatch with GHA nice?

It's basically two steps: `pipx install hatch` followed by `hatch run <command>`.

Provisioning and configuring the runtime is handled by `actions/setup-python` and some environment variables if need be. Configuring `tox-gh-actions` looks a bit more involved but definitely doable too.

**Comment by effigies** [comment]
If hatch works smoothly, that's fine with me.
